### DISCUSSION TITLE
.chezmoiexternal - support for git repositories?

### DISCUSSION THREAD

**Discussion by j-xella** [twpayne/chezmoi#1831] [Ideas]
Hi,

Currently, `.chezmoiexternal` supports files and archives. Is it possible for it to also support git repositories? So that:

archive add ~ git clone
archive update ~ git update --rebase

Here is a use case - many vim plugins these days can simply be installed via `git clone` into `~/.vim/pack/author/start/xxx` folder. It would be great if chezmoi could be configured to do this as a part of set-up. After reading the documentation, I could only come with the following solutions:

* use archives provided by github - but then I would not have access to plugin history
* create some `run_xxx.sh` script(-s) that would do it - but that would not be very declarative, and potentially require different code for different platforms, etc...

**Comment by twpayne** [comment]
Let's discuss this in issue #1838.
